src/main/java/com/puppycrawl/tools/checkstyle/DefaultLogger.java

```diff
@@ -159,30 +159,30 @@ public void addError(AuditEvent event) {
     @Override
     public void addException(AuditEvent event, Throwable throwable) {
         synchronized (errorWriter) {
-            final Violation addExceptionMessage = new Violation(1,
+            final LocalizedMessage addExceptionMessage = new LocalizedMessage(
```

Comment:
we do not need all functionality that was in Violation class.

We should have ctor:
` new LocalizedMessage(ADD_EXCEPTION_MESSAGE, new String[] {event.getFileName()});`

all other pararmeters are not required, Bundle can be set as default value in LocalizedMessage.

please remove all fields from LocalizedMessage that we do not currently use.
I see usage only of getMessage, so severity , source class, .... should be removed.
Keep only fields that used in getMessage.

Reply:
The `getMessage()` method is being used in DefaultLogger so it cannot be removed, reset methods are removed.

Reply:
@AkMo3 ,
please be attentive.
ctor should be `new LocalizedMessage(ADD_EXCEPTION_MESSAGE, new String[] {event.getFileName()});`
 not a 
`new LocalizedMessage(new String[] {event.getFileName()}, ADD_EXCEPTION_MESSAGE);`

Reply:
The parameters were swapped so that we do not need to use `varargs` as array cannot be the last element in parameters as per pmd.

Reply:
it is not pmd, it is Eclipse validation.

`Type String[] of the last argument to constructor DefaultLogger.LocalizedMessage(String, Object...) doesn't exactly match the vararg parameter type. Cast to Object[] to confirm the non-varargs invocation, or pass individual arguments of type Object for a varargs invocation.`

can you make it `new Object[] {"myfile"}` ? Or even better to udpate ctor that accept "String[]". As this is internal implementation and we know that all will be string.

Reply:
> can you make it new Object[] {"myfile"} ? Or even better to udpate ctor that accept "String[]". As this is internal implementation and we know that all will be string.

Updated to use `String[]`